bin/cron/ops_data_pull

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/properties'
+include ApplicationHelper
+include UsersHelper
+
+pd_scripts = Script.all.select { |script| !!script.pd }
+pd_scripts.each do |script|
+  # Get all users with any activity in the script
+  users = User.joins('left join user_scripts on user_scripts.user_id = users.id').where(user_scripts: {script_id: script})
+  headers = nil
+  data = users.map do |user|
+    row = {}
+    row.merge!({
+                   :'ID' => user.id,
+                   :'Ops First Name' => user.ops_first_name,
+                   :'Ops Last Name' => user.ops_last_name,
+                   :'Email' => user.email,
+                   :'District' => user.district.try(:name) || 'None'
+               })
+    user_progress = summarize_user_progress(script, user)
+    percent = percent_complete(script, user)
+    script.stages.each do |stage|
+      levels = Hash[stage.script_levels.map(&:level).map do |level|
+                      key = level.id.to_s
+                      [key, (progress = user_progress[:levels][level.id]) && progress[:status] == 'perfect' ? '1' : '0']
+                    end]
+      row.merge!(levels)
+      row.merge!({:"Stage #{stage.position} Percent Complete" => percent[stage.position - 1].to_s})
+    end
+    row.merge!({:'Script Percent Complete' => percent_complete_total(script, user).to_s})
+    headers ||= row.keys
+    row.values
+  end
+  locals_options = {headers: headers, data: data, script: {name: script.name, id: script.id}}
+  Properties.set("pd_progress_#{script.id}", locals_options)
```

Comment:
maybe include a timestamp in here so that the users can tell how up to date or not the data is (and so it's easier to tell if the script is failing to run for some reason).